app/src/main/java/org/dhis2/utils/granularsync/GranularSyncPresenterImpl.kt

```diff
@@ -481,4 +484,27 @@ class GranularSyncPresenterImpl(
 
     override fun displayMessage(message: String?) {
     }
+
+    override fun syncErrors(): List<ErrorViewModel> {
+        return arrayListOf<ErrorViewModel>().apply {
+            addAll(
+                errorMapper.mapD2Error(
+                    d2.maintenanceModule().d2Errors().blockingGet()
+                )
+            )
+            addAll(
+                errorMapper.mapConflict(
+                    d2.importModule().trackerImportConflicts().blockingGet()
+                )
+            )
+            addAll(
+                errorMapper.mapFKViolation(
+                    d2.maintenanceModule().foreignKeyViolations().blockingGet()
+                )
+            )
+            sortByDescending {
+                it.creationDate?.time
+            }
+        }
+    }
```

Comment:
I'm approving it ! but I would have put it simpler these lines, though

val list1 = Z
val list2 = Y                
val list3 = X

return whateverTypeListOf(X,Y,Z). sortByDescending {it.creationDate?.time}

Reply:
that won't work. arrayListOf does not accept lists as arguments and sortByDescending returns a Unit.